pkg/sfu/forwarder.go

```diff
@@ -396,43 +416,41 @@ func (f *Forwarder) Allocate(availableChannelCapacity int64, allowPause bool, br
 	switch {
 	case f.muted:
 		state = VideoAllocationStateMuted
-	case optimalBandwidthNeeded == 0:
-		if len(f.availableLayers) == 0 {
-			// feed is dry
-			state = VideoAllocationStateFeedDry
+	case len(f.availableLayers) == 0:
+		// feed is dry
+		state = VideoAllocationStateFeedDry
+	case !bitrateAvailable:
+		// feed bitrate not yet calculated for all available layers
+		state = VideoAllocationStateAwaitingMeasurement
+
+		if availableChannelCapacity == ChannelCapacityInfinity {
+			//
+			// Channel capacity allows a free pass.
+			// So, resume with the highest layer available <= max subscribed layer
+			// If already resumed, move allocation to the highest available layer <= max subscribed layer
+			//
+			targetLayers.spatial = int32(math.Min(float64(f.maxLayers.spatial), float64(f.availableLayers[len(f.availableLayers)-1])))
+			targetLayers.temporal = int32(math.Max(0, float64(f.maxLayers.temporal)))
+
+			if f.targetLayers == InvalidLayers {
+				change = VideoStreamingChangeResuming
+			}
 		} else {
-			// feed bitrate is not yet calculated
-			state = VideoAllocationStateAwaitingMeasurement
-
-			if availableChannelCapacity == ChannelCapacityInfinity {
-				//
-				// Channel capacity allows a free pass.
-				// So, resume with the highest layer available <= max subscribed layer
-				// If already resumed, move allocation to the highest available layer <= max subscribed layer
-				//
-				targetLayers.spatial = int32(math.Min(float64(f.maxLayers.spatial), float64(f.availableLayers[len(f.availableLayers)-1])))
-				targetLayers.temporal = int32(math.Max(0, float64(f.maxLayers.temporal)))
+			if !allowPause {
+				// when pause is disallowed, pick the lowest available
+				targetLayers.spatial = int32(math.Min(float64(f.maxLayers.spatial), float64(f.availableLayers[0])))
+				targetLayers.temporal = int32(math.Min(0, float64(f.maxLayers.temporal)))
 
 				if f.targetLayers == InvalidLayers {
 					change = VideoStreamingChangeResuming
 				}
 			} else {
-				if !allowPause {
-					// when pause is disallowed, pick the lowest available
-					targetLayers.spatial = int32(math.Min(float64(f.maxLayers.spatial), float64(f.availableLayers[0])))
-					targetLayers.temporal = int32(math.Min(0, float64(f.maxLayers.temporal)))
-
-					if f.targetLayers == InvalidLayers {
-						change = VideoStreamingChangeResuming
-					}
-				} else {
-					// disable forwarding as it is not known how big this stream is
-					// and if it will fit in the available channel capacity
-					state = VideoAllocationStateDeficient
+				// disable forwarding as it is not known how big this stream is
+				// and if it will fit in the available channel capacity
+				state = VideoAllocationStateDeficient
 
-					if f.targetLayers != InvalidLayers {
-						change = VideoStreamingChangePausing
-					}
+				if f.targetLayers != InvalidLayers {
```

Comment:
Can we trust our BWE enough to pause? I'm concerned it'll pause unnecessarily due to inaccurate BWE prints.

Seems better to lean on the side of non-congestion since it's more likely.. wdyt?

Reply:
Yeah. Going to re-do this function as this is used in only one scenario, i. e. allocate the best available layer. Will clean up this function so that it is not confusing.